File: src/main/java/com/baulsupp/oksocial/util/InetAddressParam.java
package com.baulsupp.oksocial.util;

import java.net.InetSocketAddress;

public class InetAddressParam {
  public InetSocketAddress address;

  public InetAddressParam(String hostAndPort) {
    String[] parts = hostAndPort.split(":");
    this.address = InetSocketAddress.createUnresolved(parts[0], Integer.parseInt(parts[1]));
  }
}

File: src/main/kotlin/com/baulsupp/okurl/util/InetAddressParam.kt
package com.baulsupp.okurl.util

import java.net.InetSocketAddress

class InetAddressParam(hostAndPort: String) {
  var address: InetSocketAddress

  init {
    val parts = hostAndPort.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    this.address = InetSocketAddress.createUnresolved(parts[0], Integer.parseInt(parts[1]))
  }
}
